Enable sparse_csr.to_dense() for bool, float16, bfloat16 and complex

IvanYashchuk opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #60838 Add cuSPARSE descriptors
* **#60657 Enable sparse_csr.to_dense() for bool, float16, bfloat16 and complex**

Fixes https://github.com/pytorch/pytorch/issues/60648

Differential Revision: [D29408102](https://our.internmc.facebook.com/intern/diff/D29408102)

IvanYashchuk committed:
Enable sparse_csr.to_dense() for bool, float16, bfloat16 and complex

Fixes https://github.com/pytorch/pytorch/issues/60648

[ghstack-poisoned]

IvanYashchuk committed:
Update on "Enable sparse_csr.to_dense() for bool, float16, bfloat16 and complex"


Fixes https://github.com/pytorch/pytorch/issues/60648

[ghstack-poisoned]

IvanYashchuk committed:
Update on "Enable sparse_csr.to_dense() for bool, float16, bfloat16 and complex"


Fixes https://github.com/pytorch/pytorch/issues/60648

[ghstack-poisoned]

cpuhrsch reviewed (approved):
LGTM!

cpuhrsch commented:
@cpuhrsch has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29408102).

IvanYashchuk commented:
@cpuhrsch, is anything blocking this PR, could you please merge it?

IvanYashchuk committed:
Update on "Enable sparse_csr.to_dense() for bool, float16, bfloat16 and complex"


Fixes https://github.com/pytorch/pytorch/issues/60648

Differential Revision: [D29408102](https://our.internmc.facebook.com/intern/diff/D29408102)

[ghstack-poisoned]
